components/modal/Register/register.component.js

```diff
@@ -1,214 +1,216 @@
-import React, { Component } from 'react';
-import { Form, Input, Label, FormGroup, FormFeedback } from 'reactstrap';
-import { isEmail, isDecimal } from 'validator';
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { isEmail } from 'validator';
+import PropTypes from 'prop-types';
 import classNames from './register.component.module.scss';
-class Register extends Component {
 
-    constructor(props) {
-        super(props);
-        this.state = this.getInitialState(props);
+function Register(props) {
+  const { register, handleSubmit, errors } = useForm();
+  const getMembersIntroURL = (RDSID) => `https://api.realdevsquad.com/members/intro/${RDSID}`;
+  const parameters = props.rdsusername;
+  const onSubmit = (data) => {
+    console.log(data);
+    function charRem(paramstr) {
+      return paramstr.replace('@', '');
     }
 
-    getInitialState = (props) => ({
-        recruiterData: {
-            param: props.rdsusername,
-            company: '',
-            first_name: '',
-            last_name: '',
-            designation: '',
-            reason: '',
-            email: '',
-            currency: '',
-            package: ''
-        },
-        errors: {},
-    });
-
-    getMembersIntroURL = (rdsId) => `https://api.realdevsquad.com/members/${rdsId}`;
-
-    handleSubmit = (e) => {
-        e.preventDefault();
-        const { recruiterData } = this.state;
-        const errors = this.validate();
-
-        function charRem(paramstr) {
-            return paramstr.replace('@', '');
-        }
-
-        const parameter = charRem(recruiterData.param);
-        const rdsApiURL = this.getMembersIntroURL(parameter);
-        console.log(rdsApiURL);
-
-        if (Object.keys(errors).length === 0) {
-            console.log(recruiterData);
-            fetch(rdsApiURL, {
-                method: "POST",
-                body: JSON.stringify(this.state),
-                headers: {
-                    'Accept': 'application/json',
-                    'Content-Type': 'application/json'
-                },
-            }).then(
-                (response) => (response.json())
-            ).then((response) => {
-                if (response.status === 'success') {
-                    alert("Message Sent.");
-                    this.resetForm()
-                } else if (response.status === 'fail') {
-                    alert("Message failed to send.")
-                }
-            })
+    const parameter = charRem(parameters);
+    const rdsApiURL = getMembersIntroURL(parameter);
+    console.log(rdsApiURL);
```

Comment:
This can be removed for the production code 😅

Reply:
Definitely haha